In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '잠실역 8번출구']
df

자치구      대여소명  대여소번호     년   월  시간     lambda         mu     c  \
567072  송파구  잠실역 8번출구   1295  2020   6   0   2.300000   2.433333  16.0   
567073  송파구  잠실역 8번출구   1295  2020   6   1   1.200000   1.500000  16.0   
567074  송파구  잠실역 8번출구   1295  2020   6   2   1.133333   1.233333  16.0   
567075  송파구  잠실역 8번출구   1295  2020   6   3   0.400000   0.266667  16.0   
567076  송파구  잠실역 8번출구   1295  2020   6   4   0.533333   0.233333  16.0   
567077  송파구  잠실역 8번출구   1295  2020   6   5   1.766667   0.233333  16.0   
567078  송파구  잠실역 8번출구   1295  2020   6   6   2.133333   1.133333  16.0   
567079  송파구  잠실역 8번출구   1295  2020   6   7   6.933333   1.933333  16.0   
567080  송파구  잠실역 8번출구   1295  2020   6   8  16.733333   4.033333  16.0   
567081  송파구  잠실역 8번출구   1295  2020   6   9   7.866667   3.333333  16.0   
567082  송파구  잠실역 8번출구   1295  2020   6  10   5.033333   4.166667  16.0   
567083  송파구  잠실역 8번출구   1295  2020   6  11   5.200000   5.966667  16.0   
567084  송파구  잠실역 8번출구   1295  2020   6  12   7.433333   7.300000  16.0   
567085  송파구  잠실역 8번출구   1295  2020   6  13   6.833333   6.533333  16.0   
567086  송파구  잠실역 8번출구   1295  2020   6  14   7.400000  10.333333  16.0   
567087  송파구  잠실역 8번출구   1295  2020   6  15   7.566667  11.633333  16.0   
567088  송파구  잠실역 8번출구   1295  2020   6  16  10.533333  15.633333  16.0   
567089  송파구  잠실역 8번출구   1295  2020   6  17  10.266667  19.600000  16.0   
567090  송파구  잠실역 8번출구   1295  2020   6  18  12.233333  12.933333  16.0   
567091  송파구  잠실역 8번출구   1295  2020   6  19  12.600000  12.533333  16.0   
567092  송파구  잠실역 8번출구   1295  2020   6  20  10.633333  10.633333  16.0   
567093  송파구  잠실역 8번출구   1295  2020   6  21   9.800000  10.166667  16.0   
567094  송파구  잠실역 8번출구   1295  2020   6  22   6.900000   6.733333  16.0   
567095  송파구  잠실역 8번출구   1295  2020   6  23   4.500000   4.500000  16.0   
567096  송파구  잠실역 8번출구   1295  2020   7   0   2.400000   2.533333  16.0   
567097  송파구  잠실역 8번출구   1295  2020   7   1   1.233333   1.400000  16.0   
567098  송파구  잠실역 8번출구   1295  2020   7   2   1.133333   0.966667  16.0   
567099  송파구  잠실역 8번출구   1295  2020   7   3   0.633333   0.466667  16.0   
567100  송파구  잠실역 8번출구   1295  2020   7   4   0.766667   0.566667  16.0   
567101  송파구  잠실역 8번출구   1295  2020   7   5   1.066667   0.566667  16.0   
567102  송파구  잠실역 8번출구   1295  2020   7   6   2.466667   1.033333  16.0   
567103  송파구  잠실역 8번출구   1295  2020   7   7   5.666667   1.733333  16.0   
567104  송파구  잠실역 8번출구   1295  2020   7   8  10.433333   3.800000  16.0   
567105  송파구  잠실역 8번출구   1295  2020   7   9   6.733333   2.866667  16.0   
567106  송파구  잠실역 8번출구   1295  2020   7  10   3.266667   3.966667  16.0   
567107  송파구  잠실역 8번출구   1295  2020   7  11   4.866667   6.666667  16.0   
567108  송파구  잠실역 8번출구   1295  2020   7  12   6.333333   6.766667  16.0   
567109  송파구  잠실역 8번출구   1295  2020   7  13   5.966667   6.866667  16.0   
567110  송파구  잠실역 8번출구   1295  2020   7  14   7.833333  10.133333  16.0   
567111  송파구  잠실역 8번출구   1295  2020   7  15   7.700000  10.833333  16.0   
567112  송파구  잠실역 8번출구   1295  2020   7  16   8.033333  12.333333  16.0   
567113  송파구  잠실역 8번출구   1295  2020   7  17   9.833333  12.800000  16.0   
567114  송파구  잠실역 8번출구   1295  2020   7  18  11.300000  11.500000  16.0   
567115  송파구  잠실역 8번출구   1295  2020   7  19  11.300000  11.166667  16.0   
567116  송파구  잠실역 8번출구   1295  2020   7  20   8.500000   8.866667  16.0   
567117  송파구  잠실역 8번출구   1295  2020   7  21   8.700000   8.800000  16.0   
567118  송파구  잠실역 8번출구   1295  2020   7  22   7.000000   6.700000  16.0   
567119  송파구  잠실역 8번출구   1295  2020   7  23   4.300000   4.733333  16.0   
567120  송파구  잠실역 8번출구   1295  2020   8   0   1.166667   1.266667  16.0   
567121  송파구  잠실역 8번출구   1295  2020   8   1   1.266667   1.233333  16.0   
567122  송파구  잠실역 8번출구   1295  2020   8   2   0.733333   0.766667  16.0   
567123  송파구  잠실역 8번출구   1295  2020   8   3   0.666667   0.566667  16.0   
567124  송파구  잠실역 8번출구   1295  2020   8   4   0.466667   0.300000  16.0   
567125  송파구 

In [4]:
df = df.drop(['대여소번호', 'c', 'rho', 'p_c', 'p_0', 'p_1'], axis = 1)
df

자치구      대여소명     년   월  시간     lambda         mu      p_1+  \
567072  송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   
567073  송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   
567074  송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   
567075  송파구  잠실역 8번출구  2020   6   3   0.400000   0.266667  0.999492   
567076  송파구  잠실역 8번출구  2020   6   4   0.533333   0.233333  0.999999   
567077  송파구  잠실역 8번출구  2020   6   5   1.766667   0.233333  1.000000   
567078  송파구  잠실역 8번출구  2020   6   6   2.133333   1.133333  0.999981   
567079  송파구  잠실역 8번출구  2020   6   7   6.933333   1.933333  1.000000   
567080  송파구  잠실역 8번출구  2020   6   8  16.733333   4.033333  1.000000   
567081  송파구  잠실역 8번출구  2020   6   9   7.866667   3.333333  0.999999   
567082  송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275   
567083  송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   
567084  송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   
567085  송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   
567086  송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   
567087  송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196   
567088  송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377   
567089  송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802   
567090  송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   
567091  송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   
567092  송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   
567093  송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   
567094  송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   
567095  송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   
567096  송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   
567097  송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   
567098  송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632   
567099  송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002   
567100  송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918   
567101  송파구  잠실역 8번출구  2020   7   5   1.066667   0.566667  0.999981   
567102  송파구  잠실역 8번출구  2020   7   6   2.466667   1.033333  0.999999   
567103  송파구  잠실역 8번출구  2020   7   7   5.666667   1.733333  1.000000   
567104  송파구  잠실역 8번출구  2020   7   8  10.433333   3.800000  1.000000   
567105  송파구  잠실역 8번출구  2020   7   9   6.733333   2.866667  0.999999   
567106  송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   
567107  송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   
567108  송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   
567109  송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687   
567110  송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137   
567111  송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894   
567112  송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113   
567113  송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   
567114  송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   
567115  송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   
567116  송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   
567117  송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   
567118  송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   
567119  송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   
567120  송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145   
567121  송파구  잠실역 8번출구  2020   8   1   1.266667   1.233333  0.952881   
567122  송파구  잠실역 8번출구  2020   8   2   0.733333   0.766667  0.918013   
567123  송파구  잠실역 8번출구  2020   8   3   0.666667   0.566667  0.988112   
567124  송파구  잠실역 8번출구  2020   8   4   0.466667   0.300000  0.999696   
567125  송파구  잠실역 8번출구  2020   8   5   0.533333   0.266667  0.999992   
567126  송파구  잠실역 8번출구  2020   8   6   1.300000   0.633333  0.999995   
567127  송파구  잠실역 8번출구  2020   8 

In [5]:
data = pd.read_csv('./잠실역 8번출구 균형, 불균형.csv', encoding = 'cp949')
data

자치구      대여소명     년   월  시간        재고량     3사분위수     2사분위수  1사분위수   균형
0    송파구  잠실역 8번출구  2020   6   0  -6.766667  8.566667  3.416667 -1.675  불균형
1    송파구  잠실역 8번출구  2020   6   1  -6.933333  8.566667  3.416667 -1.675  불균형
2    송파구  잠실역 8번출구  2020   6   2  -6.800000  8.566667  3.416667 -1.675  불균형
3    송파구  잠실역 8번출구  2020   6   3  -6.566667  8.566667  3.416667 -1.675  불균형
4    송파구  잠실역 8번출구  2020   6   4  -6.266667  8.566667  3.416667 -1.675  불균형
5    송파구  잠실역 8번출구  2020   6   5  -4.733333  8.566667  3.416667 -1.675  불균형
6    송파구  잠실역 8번출구  2020   6   6  -3.633333  8.566667  3.416667 -1.675  불균형
7    송파구  잠실역 8번출구  2020   6   7   1.366667  8.566667  3.416667 -1.675   균형
8    송파구  잠실역 8번출구  2020   6   8   9.833333  8.566667  3.416667 -1.675  불균형
9    송파구  잠실역 8번출구  2020   6   9  14.400000  8.566667  3.416667 -1.675  불균형
10   송파구  잠실역 8번출구  2020   6  10  15.500000  8.566667  3.416667 -1.675  불균형
11   송파구  잠실역 8번출구  2020   6  11  14.733333  8.566667  3.416667 -1.675  불균형
12   송파구  잠실역 8번출구  2020   6  12  14.900000  8.566667  3.416667 -1.675  불균형
13   송파구  잠실역 8번출구  2020   6  13  15.233333  8.566667  3.416667 -1.675  불균형
14   송파구  잠실역 8번출구  2020   6  14  12.300000  8.566667  3.416667 -1.675  불균형
15   송파구  잠실역 8번출구  2020   6  15   8.233333  8.566667  3.416667 -1.675   균형
16   송파구  잠실역 8번출구  2020   6  16   3.133333  8.566667  3.416667 -1.675   균형
17   송파구  잠실역 8번출구  2020   6  17  -6.200000  8.566667  3.416667 -1.675  불균형
18   송파구  잠실역 8번출구  2020   6  18  -6.866667  8.566667  3.416667 -1.675  불균형
19   송파구  잠실역 8번출구  2020   6  19  -6.800000  8.566667  3.416667 -1.675  불균형
20   송파구  잠실역 8번출구  2020   6  20  -6.700000  8.566667  3.416667 -1.675  불균형
21   송파구  잠실역 8번출구  2020   6  21  -7.000000  8.566667  3.416667 -1.675  불균형
22   송파구  잠실역 8번출구  2020   6  22  -6.800000  8.566667  3.416667 -1.675  불균형
23   송파구  잠실역 8번출구  2020   6  23  -6.766667  8.566667  3.416667 -1.675  불균형
24   송파구  잠실역 8번출구  2020   7   0  -4.633333  8.566667  3.416667 -1.675  불균형
25   송파구  잠실역 8번출구  2020   7   1  -4.633333  8.566667  3.416667 -1.675  불균형
26   송파구  잠실역 8번출구  2020   7   2  -4.433333  8.566667  3.416667 -1.675  불균형
27   송파구  잠실역 8번출구  2020   7   3  -4.133333  8.566667  3.416667 -1.675  불균형
28   송파구  잠실역 8번출구  2020   7   4  -3.733333  8.566667  3.416667 -1.675  불균형
29   송파구  잠실역 8번출구  2020   7   5  -3.200000  8.566667  3.416667 -1.675  불균형
30   송파구  잠실역 8번출구  2020   7   6  -1.766667  8.566667  3.416667 -1.675  불균형
31   송파구  잠실역 8번출구  2020   7   7   2.166667  8.566667  3.416667 -1.675   균형
32   송파구  잠실역 8번출구  2020   7   8   7.333333  8.566667  3.416667 -1.675   균형
33   송파구  잠실역 8번출구  2020   7   9  11.233333  8.566667  3.416667 -1.675  불균형
34   송파구  잠실역 8번출구  2020   7  10  10.666667  8.566667  3.416667 -1.675  불균형
35   송파구  잠실역 8번출구  2020   7  11   8.933333  8.566667  3.416667 -1.675  불균형
36   송파구  잠실역 8번출구  2020   7  12   8.633333  8.566667  3.416667 -1.675  불균형
37   송파구  잠실역 8번출구  2020   7  13   7.733333  8.566667  3.416667 -1.675   균형
38   송파구  잠실역 8번출구  2020   7  14   5.433333  8.566667  3.416667 -1.675   균형
39   송파구  잠실역 8번출구  2020   7  15   2.566667  8.566667  3.416667 -1.675   균형
40   송파구  잠실역 8번출구  2020   7  16  -1.666667  8.566667  3.416667 -1.675   균형
41   송파구  잠실역 8번출구  2020   7  17  -4.600000  8.566667  3.416667 -1.675  불균형
42   송파구  잠실역 8번출구  2020   7  18  -4.733333  8.566667  3.416667 -1.675  불균형
43   송파구  잠실역 8번출구  2020   7  19  -4.600000  8.566667  3.416667 -1.675  불균형
44   송파구  잠실역 8번출구  2020   7  20  -4.933333  8.566667  3.416667 -1.675  불균형
45   송파구  잠실역 8번출구  2020   7  21  -5.000000  8.566667  3.416667 -1.675  불균형
46   송파구  잠실역 8번출구  2020   7  22  -4.700000  8.566667  3.416667 -1.675  불균형
47   송파구  잠실역 8번출구  2020   7  23  -5.133333  8.566667  3.416667 -1.675  불균형
48   송파구  잠실역 8번출구  2020   8   0   0.700000  8.566667  3.416667 -1.675   균형
49   송파구  잠실역 8번출구  2020   8   1   0.800000  8.566667  3.416667 -1.675   균형
50   송파구  잠실역 8번출구  2020   8   2   1.033333  8.566667  3.416667 -1.675   균형
51   송파구  잠실역 8번출구  2020   8   3   1.333333  8.566667

In [6]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구      대여소명     년   월  시간     lambda         mu      p_1+        L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   6.037473   
3    송파구  잠실역 8번출구  2020   6   3   0.400000   0.266667  0.999492  14.017272   
4    송파구  잠실역 8번출구  2020   6   4   0.533333   0.233333  0.999999  15.222236   
5    송파구  잠실역 8번출구  2020   6   5   1.766667   0.233333  1.000000  15.847826   
6    송파구  잠실역 8번출구  2020   6   6   2.133333   1.133333  0.999981  14.867030   
7    송파구  잠실역 8번출구  2020   6   7   6.933333   1.933333  1.000000  15.613333   
8    송파구  잠실역 8번출구  2020   6   8  16.733333   4.033333  1.000000  15.682415   
9    송파구  잠실역 8번출구  2020   6   9   7.866667   3.333333  0.999999  15.264714   
10   송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275  11.905418   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   8.433715   
13   송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   9.067134   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   2.464276   
15   송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196   1.849302   
16   송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377   2.044671   
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802   1.099714   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   7.124128   
22   송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   8.585138   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   5.170779   
26   송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632  11.419437   
27   송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002  13.295112   
28   송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918  13.266958   
29   송파구  잠실역 8번출구  2020   7   5   1.066667   0.566667  0.999981  14.867030   
30   송파구  잠실역 8번출구  2020   7   6   2.466667   1.033333  0.999999  15.279076   
31   송파구  잠실역 8번출구  2020   7   7   5.666667   1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333   3.800000  1.000000  15.427136   
33   송파구  잠실역 8번출구  2020   7   9   6.733333   2.866667  0.999999  15.258629   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   6.444278   
37   송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687   4.911651   
38   송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137   3.189403   
39   송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894   2.406025   
40   송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113   1.856586   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   7.725884   
46   송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   9.041641   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   5.792218   
48   송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145   6.087940   
49   송파구  잠실역 8번출구  2020   8   1   1.266667   1.233333 

In [7]:
df_new['내 균형'] = '균형'
df_new

자치구      대여소명     년   월  시간     lambda         mu      p_1+        L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   6.037473   
3    송파구  잠실역 8번출구  2020   6   3   0.400000   0.266667  0.999492  14.017272   
4    송파구  잠실역 8번출구  2020   6   4   0.533333   0.233333  0.999999  15.222236   
5    송파구  잠실역 8번출구  2020   6   5   1.766667   0.233333  1.000000  15.847826   
6    송파구  잠실역 8번출구  2020   6   6   2.133333   1.133333  0.999981  14.867030   
7    송파구  잠실역 8번출구  2020   6   7   6.933333   1.933333  1.000000  15.613333   
8    송파구  잠실역 8번출구  2020   6   8  16.733333   4.033333  1.000000  15.682415   
9    송파구  잠실역 8번출구  2020   6   9   7.866667   3.333333  0.999999  15.264714   
10   송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275  11.905418   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   8.433715   
13   송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   9.067134   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   2.464276   
15   송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196   1.849302   
16   송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377   2.044671   
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802   1.099714   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   7.124128   
22   송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   8.585138   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   5.170779   
26   송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632  11.419437   
27   송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002  13.295112   
28   송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918  13.266958   
29   송파구  잠실역 8번출구  2020   7   5   1.066667   0.566667  0.999981  14.867030   
30   송파구  잠실역 8번출구  2020   7   6   2.466667   1.033333  0.999999  15.279076   
31   송파구  잠실역 8번출구  2020   7   7   5.666667   1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333   3.800000  1.000000  15.427136   
33   송파구  잠실역 8번출구  2020   7   9   6.733333   2.866667  0.999999  15.258629   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   6.444278   
37   송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687   4.911651   
38   송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137   3.189403   
39   송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894   2.406025   
40   송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113   1.856586   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   7.725884   
46   송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   9.041641   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   5.792218   
48   송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145   6.087940   
49   송파구  잠실역 8번출구  2020   8   1   1.266667   1.233333 

# 일단 L_s, 기존균형 비교

## L_s가 Q1보다 작거나, Q3보다 크면 불균형

### L_s가 q1보다 작지만 이때 p_1+이 충분히 크다면 균형
- ls가 q1보다 작고 p_1+가 0.5보다 작으면 불균형

In [8]:
# ls가 q1보다 작으므로 불균형

e = df_new[(df_new['L_s'] < df_new['1사분위수']) & (df_new['p_1+'] < 0.6)]
e

Empty DataFrame
Columns: [자치구, 대여소명, 년, 월, 시간, lambda, mu, p_1+, L_s, W_s, 재고량, 3사분위수, 2사분위수, 1사분위수, 균형, 내 균형]
Index: []

In [9]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] < df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [10]:
# ls가 q3보다 크므로 불균형

r = df_new[(df_new['L_s'] > df_new['3사분위수']) & (df_new['L_s'] > (20*0.7))]
r

자치구      대여소명     년   월  시간     lambda        mu      p_1+        L_s  \
3    송파구  잠실역 8번출구  2020   6   3   0.400000  0.266667  0.999492  14.017272   
4    송파구  잠실역 8번출구  2020   6   4   0.533333  0.233333  0.999999  15.222236   
5    송파구  잠실역 8번출구  2020   6   5   1.766667  0.233333  1.000000  15.847826   
6    송파구  잠실역 8번출구  2020   6   6   2.133333  1.133333  0.999981  14.867030   
7    송파구  잠실역 8번출구  2020   6   7   6.933333  1.933333  1.000000  15.613333   
8    송파구  잠실역 8번출구  2020   6   8  16.733333  4.033333  1.000000  15.682415   
9    송파구  잠실역 8번출구  2020   6   9   7.866667  3.333333  0.999999  15.264714   
29   송파구  잠실역 8번출구  2020   7   5   1.066667  0.566667  0.999981  14.867030   
30   송파구  잠실역 8번출구  2020   7   6   2.466667  1.033333  0.999999  15.279076   
31   송파구  잠실역 8번출구  2020   7   7   5.666667  1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333  3.800000  1.000000  15.427136   
33   송파구  잠실역 8번출구  2020   7   9   6.733333  2.866667  0.999999  15.258629   
52   송파구  잠실역 8번출구  2020   8   4   0.466667  0.300000  0.999696  14.209303   
53   송파구  잠실역 8번출구  2020   8   5   0.533333  0.266667  0.999992  15.000130   
54   송파구  잠실역 8번출구  2020   8   6   1.300000  0.633333  0.999995  15.050083   
55   송파구  잠실역 8번출구  2020   8   7   3.433333  1.100000  1.000000  15.528571   
56   송파구  잠실역 8번출구  2020   8   8   5.700000  2.366667  1.000000  15.290006   
57   송파구  잠실역 8번출구  2020   8   9   4.200000  2.466667  0.999917  14.578924   
76   송파구  잠실역 8번출구  2020   9   4   0.666667  0.366667  0.999968  14.778433   
77   송파구  잠실역 8번출구  2020   9   5   0.900000  0.233333  1.000000  15.650000   
78   송파구  잠실역 8번출구  2020   9   6   1.300000  0.600000  0.999998  15.142890   
79   송파구  잠실역 8번출구  2020   9   7   6.533333  1.433333  1.000000  15.718954   
80   송파구  잠실역 8번출구  2020   9   8  10.566667  3.366667  1.000000  15.532407   
81   송파구  잠실역 8번출구  2020   9   9   7.866667  3.033333  1.000000  15.372415   
99   송파구  잠실역 8번출구  2020  10   3   0.733333  0.466667  0.999737  14.257828   
101  송파구  잠실역 8번출구  2020  10   5   1.766667  0.133333  1.000000  15.918367   
103  송파구  잠실역 8번출구  2020  10   7   6.400000  1.966667  1.000000  15.556391   
104  송파구  잠실역 8번출구  2020  10   8  11.300000  3.300000  1.000000  15.587500   
105  송파구  잠실역 8번출구  2020  10   9   8.333333  3.333333  1.000000  15.333336   
125  송파구  잠실역 8번출구  2020  11   5   0.800000  0.133333  1.000000  15.800000   
126  송파구  잠실역 8번출구  2020  11   6   1.200000  0.433333  1.000000  15.434783   
127  송파구  잠실역 8번출구  2020  11   7   4.766667  0.933333  1.000000  15.756522   
128  송파구  잠실역 8번출구  2020  11   8   8.800000  2.133333  1.000000  15.680000   
129  송파구  잠실역 8번출구  2020  11   9   5.300000  2.000000  1.000000  15.393940   
144  송파구  잠실역 8번출구  2020  12   0   0.566667  0.366667  0.999667  14.177060   
148  송파구  잠실역 8번출구  2020  12   4   0.133333  0.066667  0.999992  15.000130   
149  송파구  잠실역 8번출구  2020  12   5   0.233333  0.100000  0.999999  15.250009   
150  송파구  잠실역 8번출구  2020  12   6   0.966667  0.366667  1.000000  15.388890   
151  송파구  잠실역 8번출구  2020  12   7   1.933333  0.366667  1.000000  15.765957   
152  송파구  잠실역 8번출구  2020  12   8   4.466667  0.633333  1.000000  15.834783   
153  송파구  잠실역 8번출구  2020  12   9   3.800000  1.000000  1.000000  15.642857   
168  송파구  잠실역 8번출구  2021   1   0   0.733333  0.300000  1.000000  15.307697   
171  송파구  잠실역 8번출구  2021   1   3   0.200000  0.033333  1.000000  15.800000   
173  송파구  잠실역 8번출구  2021   1   5   0.100000  0.033333  1.000000  15.500000   
175  송파구  잠실역 8번출구  2021   1   7   1.166667  0.233333  1.000000  15.750000   
176  송파구  잠실역 8번출구  2021   1   8   3.433333  0.533333  1.000000  15.816092   
177  송파구  잠실역 8번출구  2021   1   9   2.200000  0.300000  1.000000  15.842105   
195  송파구  잠실역 8번출구  2021   2   3   0.100000  0.033333  1.000000  15.500000   
197  송파구  잠실역 8번출구  2021   2   5   0.133333  0.066667  0.999992  15.000130   
199  송파구  잠실역 8번출구  2021   2   7   0.966667  0.100000  1.000000  15.884615   
200  송파구  잠실역 8번출구  2021   

In [11]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] > df_new['3사분위수'][i]) & (df_new['L_s'][i] > (20*0.7)):
        df_new['내 균형'][i] = '불균형'

# L_s가 Q1~Q3사이이더라도 이때 p_1+가 너무 작으면(50%이하) 불균형
- 따릉이 목표 거치율이 60~70%이므로 한대라도 있을 확률인 p_1+도 유사하게 잡기 위해

In [12]:
t = df_new[(df_new['3사분위수']>=df_new['L_s']) & (df_new['L_s'] >= df_new['1사분위수'])]
t

자치구      대여소명     년   월  시간     lambda         mu      p_1+       L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097  6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393  3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660  6.037473   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781  4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306  8.433715   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153  2.464276   
15   송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196  1.849302   
16   송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377  2.044671   
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802  1.099714   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517  6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640  8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176  8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346  7.124128   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176  8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447  6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342  5.170779   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776  4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712  2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180  6.444278   
37   송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687  4.911651   
38   송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137  3.189403   
39   송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894  2.406025   
40   송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113  1.856586   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579  3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562  7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595  8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271  6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646  7.725884   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205  5.792218   
48   송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145  6.087940   
50   송파구  잠실역 8번출구  2020   8   2   0.733333   0.766667  0.918013  6.943210   
58   송파구  잠실역 8번출구  2020   8  10   2.600000   3.000000  0.853834  4.863773   
59   송파구  잠실역 8번출구  2020   8  11   3.200000   4.066667  0.783199  3.398279   
60   송파구  잠실역 8번출구  2020   8  12   3.500000   3.766667  0.900706  6.282024   
61   송파구  잠실역 8번출구  2020   8  13   3.733333   3.800000  0.932483  7.575852   
62   송파구  잠실역 8번출구  2020   8  14   4.033333   4.400000  0.892080  5.984369   
63   송파구  잠실역 8번출구  2020   8  15   3.533333   5.100000  0.692210  2.222073   
64   송파구  잠실역 8번출구  2020   8  16   4.033333   6.200000  0.650304  1.850152   
65   송파구  잠실역 8번출구  2020   8  17   4.700000   5.500000  0.843748  4.613041   
66   송파구  잠실역 8번출구  2020   8  18   5.566667   5.533333  0.943960  8.144119   
67   송파구  잠실역 8번출구  2020   8  19   5.166667   5.433333  0.914635  6.806730   
68   송파구  잠실역 8번출구  2020   8  20   5.333333   6.100000  0.860050  5.026697   
69   송파구  잠실역 8번출구  2020   8  21   3.566667   3.666667  0.927282  7.338803   
72   송파구  잠실역 8번출구  2020   9   0   2.933333   2.933333  0.941176  8.000000   
73   송파구  잠실역 8번출구  2020   9   1   1.633333   1.833333  0.873101  5.391510   
83   송파구  잠실역 8번출구  2020   9  11   6.100000   7.900000  0.769308  3.176670   
84   송파구  잠실역 8번출구  2020   9  12   7.866667   9.600000  0.813114  3.942462   
85   송파구  잠실역 8번출구  2020   9  13  10.800000  11.866667  0.887407  5.830871   
86   송파구  잠실역 8번출구  2020   9  14  11.033333  13.166667  0.829530  4.285865   
87   송파구  잠실역 8번출구  2020   9  15  11.066667  14.300000  0.770959  3.202072   
88   송파구  잠실역 8번출구  2020   

In [13]:
t[t['p_1+']< 0.6]

자치구      대여소명     년   월  시간     lambda         mu      p_1+       L_s  \
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802  1.099714   
162  송파구  잠실역 8번출구  2020  12  18   4.133333   7.666667  0.539118  1.169344   
164  송파구  잠실역 8번출구  2020  12  20   2.400000   4.166667  0.575964  1.357053   
170  송파구  잠실역 8번출구  2021   1   2   0.166667   0.333333  0.499996  0.999870   
172  송파구  잠실역 8번출구  2021   1   4   0.000000   0.000000  0.000000  0.000000   
188  송파구  잠실역 8번출구  2021   1  20   1.900000   3.166667  0.599932  1.497122   
191  송파구  잠실역 8번출구  2021   1  23   0.166667   0.500000  0.333333  0.500000   
213  송파구  잠실역 8번출구  2021   2  21   1.766667   3.966667  0.445378  0.803012   
266  송파구  잠실역 8번출구  2021   5   2   0.166667   0.400000  0.416666  0.714280   
267  송파구  잠실역 8번출구  2021   5   3   0.133333   0.500000  0.266667  0.363636   

          W_s       재고량     3사분위수     2사분위수  1사분위수   균형 내 균형  
17   0.107117 -6.200000  8.566667  3.416667 -1.675  불균형   균형  
162  0.282913 -2.866667  8.566667  3.416667 -1.675  불균형   균형  
164  0.565474 -4.033333  8.566667  3.416667 -1.675  불균형   균형  
170  5.999268  3.966667  8.566667  3.416667 -1.675   균형   균형  
172  0.000000  4.133333  8.566667  3.416667 -1.675   균형   균형  
188  0.788048  3.400000  8.566667  3.416667 -1.675   균형   균형  
191  2.999999  3.433333  8.566667  3.416667 -1.675   균형   균형  
213  0.454536  2.733333  8.566667  3.416667 -1.675   균형   균형  
266  4.285681 -1.066667  8.566667  3.416667 -1.675   균형   균형  
267  2.727273 -1.100000  8.566667  3.416667 -1.675   균형   균형

- 위 결과를 보면 l_s가 q1~q3사이에 있지만 자전거가 한대라도 있을 확률이 0.5이하이므로 불균형이다
- 특히 19시, 20시의 경우 퇴근시간이므로 이시간에 불균형이면 이용에 불편을 초래함
- 또한 21년 1월 19시의 경우 불균형인데도 불구하고 기존연구방법에서는 이를 균형으로 판단

In [14]:
for i in range(len(df_new)):
    if (df_new['3사분위수'][i]>=df_new['L_s'][i]) & (df_new['L_s'][i] >= df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [15]:
df_new

자치구      대여소명     년   월  시간     lambda         mu      p_1+        L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   6.037473   
3    송파구  잠실역 8번출구  2020   6   3   0.400000   0.266667  0.999492  14.017272   
4    송파구  잠실역 8번출구  2020   6   4   0.533333   0.233333  0.999999  15.222236   
5    송파구  잠실역 8번출구  2020   6   5   1.766667   0.233333  1.000000  15.847826   
6    송파구  잠실역 8번출구  2020   6   6   2.133333   1.133333  0.999981  14.867030   
7    송파구  잠실역 8번출구  2020   6   7   6.933333   1.933333  1.000000  15.613333   
8    송파구  잠실역 8번출구  2020   6   8  16.733333   4.033333  1.000000  15.682415   
9    송파구  잠실역 8번출구  2020   6   9   7.866667   3.333333  0.999999  15.264714   
10   송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275  11.905418   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   8.433715   
13   송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   9.067134   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   2.464276   
15   송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196   1.849302   
16   송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377   2.044671   
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802   1.099714   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   7.124128   
22   송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   8.585138   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   5.170779   
26   송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632  11.419437   
27   송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002  13.295112   
28   송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918  13.266958   
29   송파구  잠실역 8번출구  2020   7   5   1.066667   0.566667  0.999981  14.867030   
30   송파구  잠실역 8번출구  2020   7   6   2.466667   1.033333  0.999999  15.279076   
31   송파구  잠실역 8번출구  2020   7   7   5.666667   1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333   3.800000  1.000000  15.427136   
33   송파구  잠실역 8번출구  2020   7   9   6.733333   2.866667  0.999999  15.258629   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   6.444278   
37   송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687   4.911651   
38   송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137   3.189403   
39   송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894   2.406025   
40   송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113   1.856586   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   7.725884   
46   송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   9.041641   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   5.792218   
48   송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145   6.087940   
49   송파구  잠실역 8번출구  2020   8   1   1.266667   1.233333 

In [16]:
len(df_new[df_new['균형'] == '불균형'])

143

In [17]:
len(df_new[df_new['내 균형'] == '불균형'])

80

In [18]:
len(df_new)

288

In [19]:
# TN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])

32

In [20]:
# TP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')])

97

In [21]:
# FN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')])

111

In [22]:
# FP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')])

48

In [23]:
# 정확도(기존방법과 내 방법의 일치율)

(len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])) / len(df_new)

0.4479166666666667

In [24]:
# 정밀도(기존 방법에서 균형으로 판단한 데이터 중 내가 균형이라고 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]))

0.6689655172413793

In [25]:
# 재현율(내 방법에서 균형으로 판단한 것 중 기존 방법에서도 균형으로 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]))

0.46634615384615385

In [26]:
a = df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]
a

자치구      대여소명     년   월  시간     lambda         mu      p_1+        L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   6.037473   
10   송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275  11.905418   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   8.433715   
13   송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   9.067134   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   2.464276   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   7.124128   
22   송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   8.585138   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   5.170779   
26   송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632  11.419437   
27   송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002  13.295112   
28   송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918  13.266958   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   6.444278   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   7.725884   
46   송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   9.041641   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   5.792218   
72   송파구  잠실역 8번출구  2020   9   0   2.933333   2.933333  0.941176   8.000000   
73   송파구  잠실역 8번출구  2020   9   1   1.633333   1.833333  0.873101   5.391510   
74   송파구  잠실역 8번출구  2020   9   2   0.933333   0.866667  0.969534   9.733095   
75   송파구  잠실역 8번출구  2020   9   3   0.566667   0.500000  0.981973  10.798493   
82   송파구  잠실역 8번출구  2020   9  10   5.433333   5.100000  0.966202   9.490845   
83   송파구  잠실역 8번출구  2020   9  11   6.100000   7.900000  0.769308   3.176670   
84   송파구  잠실역 8번출구  2020   9  12   7.866667   9.600000  0.813114   3.942462   
85   송파구  잠실역 8번출구  2020   9  13  10.800000  11.866667  0.887407   5.830871   
89   송파구  잠실역 8번출구  2020   9  17  16.233333  18.933333  0.846143   4.671130   
90   송파구  잠실역 8번출구  2020   9  18  15.700000  16.233333  0.924175   7.202547   
91   송파구  잠실역 8번출구  2020   9  19  16.200000  15.833333  0.951328   8.548065   
92   송파구  잠실역 8번출구  2020   9  20  11.500000  13.300000  0.852185   4.821660   
93   송파구  잠실역 8번출구  2020   9  21  10.033333   9.533333  0.962120   9.211621   
94   송파구  잠실역 8번출구  2020   9  22   5.900000   6.400000  0.895714   6.107431   
95   송파구  잠실역 8번출구  2020   9  23   3.666667   3.333333  0.975336  10.192903   
96   송파구  잠실역 8번출구  2020  10   0   1.633333   1.800000  0.885447   5.768006   
97   송파구  잠실역 8번출구  2020  10   1   1.300000   1.500000  0.853834   4.863773   
98   송파구  잠실역 8번출구  2020  10   2   1.166667   0.966667  0.991179  11.891419   
100  송파구  잠실역 8번출구  2020  10   4   0.666667   0.533333  0.994241  12.391625   
102  송파구  잠실역 8번출구  2020  10   6   1.433333   1.033333  0.998508  13.482176   
106  송파구  잠실역 8번출구  2020  10  10   5.766667   5.233333 

In [27]:
len(a)

111

In [28]:
len(a[a['p_1+'] > 0.9])

64

In [29]:
len(a[a['p_1+'] > 0.8])

92

In [30]:
len(a[a['p_1+'] > 0.7])

107

In [31]:
len(a[a['p_1+'] > 0.6])

111

- 기존방법이 불균형으로 판단한 데이터 중 본 연구에서 균형으로 판단한 데이터는 111개이지만
- 이 중 대여소에 적어도 자전거가 1대 있을 확률이 70%가 넘는 데이터가 107개로 96%이상의 경우에 이용자가 대여소에 도착했을 때 자전거를 빌릴 수 있는 확률이 70%이상이었다.
- 따라서 이러한 경우에는 자전거 재배치를 위한 대여소 내 재고의 균형/ 불균형을 판단할 경우에는 재배치를 할 필요가 없다고 판단하여 균형으로 판단할 수 있다.

In [32]:
b = df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]
b

자치구      대여소명     년   월  시간     lambda        mu      p_1+        L_s  \
7    송파구  잠실역 8번출구  2020   6   7   6.933333  1.933333  1.000000  15.613333   
31   송파구  잠실역 8번출구  2020   7   7   5.666667  1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333  3.800000  1.000000  15.427136   
52   송파구  잠실역 8번출구  2020   8   4   0.466667  0.300000  0.999696  14.209303   
53   송파구  잠실역 8번출구  2020   8   5   0.533333  0.266667  0.999992  15.000130   
54   송파구  잠실역 8번출구  2020   8   6   1.300000  0.633333  0.999995  15.050083   
55   송파구  잠실역 8번출구  2020   8   7   3.433333  1.100000  1.000000  15.528571   
56   송파구  잠실역 8번출구  2020   8   8   5.700000  2.366667  1.000000  15.290006   
57   송파구  잠실역 8번출구  2020   8   9   4.200000  2.466667  0.999917  14.578924   
78   송파구  잠실역 8번출구  2020   9   6   1.300000  0.600000  0.999998  15.142890   
79   송파구  잠실역 8번출구  2020   9   7   6.533333  1.433333  1.000000  15.718954   
103  송파구  잠실역 8번출구  2020  10   7   6.400000  1.966667  1.000000  15.556391   
150  송파구  잠실역 8번출구  2020  12   6   0.966667  0.366667  1.000000  15.388890   
151  송파구  잠실역 8번출구  2020  12   7   1.933333  0.366667  1.000000  15.765957   
152  송파구  잠실역 8번출구  2020  12   8   4.466667  0.633333  1.000000  15.834783   
153  송파구  잠실역 8번출구  2020  12   9   3.800000  1.000000  1.000000  15.642857   
168  송파구  잠실역 8번출구  2021   1   0   0.733333  0.300000  1.000000  15.307697   
170  송파구  잠실역 8번출구  2021   1   2   0.166667  0.333333  0.499996   0.999870   
171  송파구  잠실역 8번출구  2021   1   3   0.200000  0.033333  1.000000  15.800000   
172  송파구  잠실역 8번출구  2021   1   4   0.000000  0.000000  0.000000   0.000000   
173  송파구  잠실역 8번출구  2021   1   5   0.100000  0.033333  1.000000  15.500000   
175  송파구  잠실역 8번출구  2021   1   7   1.166667  0.233333  1.000000  15.750000   
176  송파구  잠실역 8번출구  2021   1   8   3.433333  0.533333  1.000000  15.816092   
177  송파구  잠실역 8번출구  2021   1   9   2.200000  0.300000  1.000000  15.842105   
188  송파구  잠실역 8번출구  2021   1  20   1.900000  3.166667  0.599932   1.497122   
191  송파구  잠실역 8번출구  2021   1  23   0.166667  0.500000  0.333333   0.500000   
195  송파구  잠실역 8번출구  2021   2   3   0.100000  0.033333  1.000000  15.500000   
197  송파구  잠실역 8번출구  2021   2   5   0.133333  0.066667  0.999992  15.000130   
199  송파구  잠실역 8번출구  2021   2   7   0.966667  0.100000  1.000000  15.884615   
200  송파구  잠실역 8번출구  2021   2   8   3.600000  0.766667  1.000000  15.729412   
201  송파구  잠실역 8번출구  2021   2   9   3.566667  0.566667  1.000000  15.811111   
213  송파구  잠실역 8번출구  2021   2  21   1.766667  3.966667  0.445378   0.803012   
219  송파구  잠실역 8번출구  2021   3   3   0.066667  0.033333  0.999992  15.000130   
220  송파구  잠실역 8번출구  2021   3   4   0.100000  0.033333  1.000000  15.500000   
221  송파구  잠실역 8번출구  2021   3   5   0.400000  0.033333  1.000000  15.909091   
222  송파구  잠실역 8번출구  2021   3   6   1.033333  0.400000  1.000000  15.368423   
223  송파구  잠실역 8번출구  2021   3   7   2.333333  0.400000  1.000000  15.793103   
224  송파구  잠실역 8번출구  2021   3   8   6.766667  1.533333  1.000000  15.707006   
241  송파구  잠실역 8번출구  2021   4   1   1.466667  0.833333  0.999949  14.685350   
245  송파구  잠실역 8번출구  2021   4   5   0.600000  0.100000  1.000000  15.800000   
246  송파구  잠실역 8번출구  2021   4   6   1.766667  0.266667  1.000000  15.822222   
247  송파구  잠실역 8번출구  2021   4   7   4.633333  1.166667  1.000000  15.663462   
266  송파구  잠실역 8번출구  2021   5   2   0.166667  0.400000  0.416666   0.714280   
267  송파구  잠실역 8번출구  2021   5   3   0.133333  0.500000  0.266667   0.363636   
269  송파구  잠실역 8번출구  2021   5   5   0.400000  0.133333  1.000000  15.500000   
270  송파구  잠실역 8번출구  2021   5   6   1.900000  0.366667  1.000000  15.760870   
271  송파구  잠실역 8번출구  2021   5   7   3.033333  1.033333  1.000000  15.483334   
272  송파구  잠실역 8번출구  2021   5   8   8.866667  2.833333  1.000000  15.530387   

            W_s       재고량     3사분위수     2사분위수  1사분위수  균형 내 균형  
7      8.075862  1.366667  8.566667  3.416667 -1.675  균형  불균형  
31     8.976532  2.166667  8.566667  3.416667 -1.675  

In [33]:
len(b)

48

In [34]:
len(b[(b['lambda'] > b['mu']) & (b['L_s'] > (20 * 0.7))])

41

- 기존방법이 균형으로 판단한 데이터 중 본 연구에서 불균형으로 판단한 데이터는 48개이지만
- 이 중 반납이 대여보다 많고 대여소 내 평균 자전거수(L_s)가 서울시 거치대율의 목표인 거치대수의 60~70%를 초과하는 경우가 41개로 85%이상의 경우에 불필요하게 자전거의 재고가 많다고 판단된다.
- 따라서 이러한 경우에는 이용자들의 자전거 이용에는 문제가 없지만 주변 대여소의 재고가 부족하다면 자전거 재배치의 측면에서는 재배치를 할 필요가 있다고 판단하여 불균형으로 판단할 수 있다.

In [35]:
df_new

자치구      대여소명     년   월  시간     lambda         mu      p_1+        L_s  \
0    송파구  잠실역 8번출구  2020   6   0   2.300000   2.433333  0.911097   6.667845   
1    송파구  잠실역 8번출구  2020   6   1   1.200000   1.500000  0.795393   3.608375   
2    송파구  잠실역 8번출구  2020   6   2   1.133333   1.233333  0.893660   6.037473   
3    송파구  잠실역 8번출구  2020   6   3   0.400000   0.266667  0.999492  14.017272   
4    송파구  잠실역 8번출구  2020   6   4   0.533333   0.233333  0.999999  15.222236   
5    송파구  잠실역 8번출구  2020   6   5   1.766667   0.233333  1.000000  15.847826   
6    송파구  잠실역 8번출구  2020   6   6   2.133333   1.133333  0.999981  14.867030   
7    송파구  잠실역 8번출구  2020   6   7   6.933333   1.933333  1.000000  15.613333   
8    송파구  잠실역 8번출구  2020   6   8  16.733333   4.033333  1.000000  15.682415   
9    송파구  잠실역 8번출구  2020   6   9   7.866667   3.333333  0.999999  15.264714   
10   송파구  잠실역 8번출구  2020   6  10   5.033333   4.166667  0.991275  11.905418   
11   송파구  잠실역 8번출구  2020   6  11   5.200000   5.966667  0.857781   4.966479   
12   송파구  잠실역 8번출구  2020   6  12   7.433333   7.300000  0.949306   8.433715   
13   송파구  잠실역 8번출구  2020   6  13   6.833333   6.533333  0.959903   9.067134   
14   송파구  잠실역 8번출구  2020   6  14   7.400000  10.333333  0.715153   2.464276   
15   송파구  잠실역 8번출구  2020   6  15   7.566667  11.633333  0.650196   1.849302   
16   송파구  잠실역 8번출구  2020   6  16  10.533333  15.633333  0.673377   2.044671   
17   송파구  잠실역 8번출구  2020   6  17  10.266667  19.600000  0.523802   1.099714   
18   송파구  잠실역 8번출구  2020   6  18  12.233333  12.933333  0.911517   6.684125   
19   송파구  잠실역 8번출구  2020   6  19  12.600000  12.533333  0.943640   8.127304   
20   송파구  잠실역 8번출구  2020   6  20  10.633333  10.633333  0.941176   8.000000   
21   송파구  잠실역 8번출구  2020   6  21   9.800000  10.166667  0.922346   7.124128   
22   송파구  잠실역 8번출구  2020   6  22   6.900000   6.733333  0.951973   8.585138   
23   송파구  잠실역 8번출구  2020   6  23   4.500000   4.500000  0.941176   8.000000   
24   송파구  잠실역 8번출구  2020   7   0   2.400000   2.533333  0.912447   6.720360   
25   송파구  잠실역 8번출구  2020   7   1   1.233333   1.400000  0.865342   5.170779   
26   송파구  잠실역 8번출구  2020   7   2   1.133333   0.966667  0.987632  11.419437   
27   송파구  잠실역 8번출구  2020   7   3   0.633333   0.466667  0.998002  13.295112   
28   송파구  잠실역 8번출구  2020   7   4   0.766667   0.566667  0.997918  13.266958   
29   송파구  잠실역 8번출구  2020   7   5   1.066667   0.566667  0.999981  14.867030   
30   송파구  잠실역 8번출구  2020   7   6   2.466667   1.033333  0.999999  15.279076   
31   송파구  잠실역 8번출구  2020   7   7   5.666667   1.733333  1.000000  15.559322   
32   송파구  잠실역 8번출구  2020   7   8  10.433333   3.800000  1.000000  15.427136   
33   송파구  잠실역 8번출구  2020   7   9   6.733333   2.866667  0.999999  15.258629   
34   송파구  잠실역 8번출구  2020   7  10   3.266667   3.966667  0.816776   4.016082   
35   송파구  잠실역 8번출구  2020   7  11   4.866667   6.666667  0.728712   2.622607   
36   송파구  잠실역 8번출구  2020   7  12   6.333333   6.766667  0.905180   6.444278   
37   송파구  잠실역 8번출구  2020   7  13   5.966667   6.866667  0.855687   4.911651   
38   송파구  잠실역 8번출구  2020   7  14   7.833333  10.133333  0.770137   3.189403   
39   송파구  잠실역 8번출구  2020   7  15   7.700000  10.833333  0.709894   2.406025   
40   송파구  잠실역 8번출구  2020   7  16   8.033333  12.333333  0.651113   1.856586   
41   송파구  잠실역 8번출구  2020   7  17   9.833333  12.800000  0.765579   3.120191   
42   송파구  잠실역 8번출구  2020   7  18  11.300000  11.500000  0.932562   7.579562   
43   송파구  잠실역 8번출구  2020   7  19  11.300000  11.166667  0.946595   8.284676   
44   송파구  잠실역 8번출구  2020   7  20   8.500000   8.866667  0.919271   6.995046   
45   송파구  잠실역 8번출구  2020   7  21   8.700000   8.800000  0.935646   7.725884   
46   송파구  잠실역 8번출구  2020   7  22   7.000000   6.700000  0.959504   9.041641   
47   송파구  잠실역 8번출구  2020   7  23   4.300000   4.733333  0.886205   5.792218   
48   송파구  잠실역 8번출구  2020   8   0   1.166667   1.266667  0.895145   6.087940   
49   송파구  잠실역 8번출구  2020   8   1   1.266667   1.233333 

In [36]:
df_new.to_csv('./잠실역 8번출구.csv', index = False, encoding = 'cp949')

# confusion matrix

In [23]:
len(df_new)

288

In [24]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   1   1.033333   1.300000  0.794608   
2    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   2   0.500000   0.666667  0.749940   
11   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  11  10.333333   8.333333  0.999530   
12   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  12  12.266667   8.766667  0.999977   
13   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  13  11.433333  12.033333  0.935505   
14   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  14  13.566667  15.266667  0.884895   
15   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  15  12.866667  17.133333  0.750911   
16   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  16  16.500000  20.166667  0.817640   
17   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  17  19.266667  28.900000  0.666664   
35   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  11   7.533333   7.000000  0.989718   
37   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  13  11.233333  10.133333  0.994244   
38   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  14  11.000000  15.633333  0.703614   
39   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  15  12.700000  15.433333  0.822271   
40   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  16  13.166667  19.166667  0.686951   
41   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  17  17.966667  23.166667  0.775398   
50   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   2   0.233333   0.433333  0.538462   
51   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   3   0.200000   0.233333  0.855489   
56   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   8   6.666667   8.733333  0.763265   
62   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  14   5.566667   7.033333  0.791232   
63   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  15   6.466667   7.500000  0.860325   
64   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  16   6.433333   8.366667  0.768811   
65   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  17   9.566667  11.500000  0.831072   
74   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   2   0.233333   0.433333  0.538462   
85   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  13  14.566667  14.300000  0.973692   
86   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  14  14.100000  18.833333  0.748616   
87   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  15  16.600000  18.833333  0.878286   
88   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  16  18.200000  21.666667  0.838974   
89   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  17  19.700000  25.366667  0.776463   
108  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  12  14.633333  13.066667  0.995331   
109  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  13  15.866667  16.700000  0.935477   
110  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  14  16.866667  21.466667  0.785517   
111  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  15  18.700000  19.600000  0.938294   
112  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  16  22.033333  26.733333  0.823542   
120  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   0   0.700000   0.500000  0.999977   
121  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   1   1.100000   0.833333  0.999898   
122  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   2   0.466667   0.466667  0.965517   
123  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   3   0.033333   0.233333  0.142857   
124  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   4   0.266667   0.300000  0.885115   
126  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   6   1.966667   0.333333  1.000000   
127  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   7   8.833333   3.500000  1.000000   
128  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   8  16.366667   9.500000  1.000000   
129  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   9  10.900000   6.900000  0.999999   
130  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  10   8.433333   4.100000  1.000000   
131  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  11   6.733333   4.966667  0.999948   
132  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  12   9.200000   8.600000  0.988505   
133  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  13   9.800000   9.633333  0.973156   
134  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  14  11.066667  11.700000  0.932411   
135  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  15  13.500000  12.000000  0.995754   
136  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  16  13.666667  17.500000  0.780784   
137  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  17  14.700000  21.833333  0.673279   
138  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  18  16.400000  25.033333  0.655125   
1

In [25]:
len(z)

89

In [26]:
# TP

len(z[z['균형'] == '균형'])

34

In [27]:
# FN

len(z[z['균형'] == '불균형'])

55

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 55개

- 균형인데 균형이라고 판단한 수가 34개

In [28]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   0   1.666667   1.533333  0.991495   
3    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   3   0.366667   0.333333  0.993272   
4    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   4   0.833333   0.466667  1.000000   
5    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   5   1.200000   0.300000  1.000000   
6    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   6   2.933333   1.266667  1.000000   
7    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   7  10.200000   5.666667  1.000000   
8    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   8  19.033333  17.500000  0.991595   
9    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   9  17.633333   9.533333  1.000000   
10   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  10   9.733333   5.800000  1.000000   
18   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  18  27.400000  30.466667  0.894477   
19   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  19  17.966667  18.433333  0.951743   
20   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  20  14.533333  14.766667  0.957284   
21   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  21  11.633333  11.666667  0.964118   
22   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  22   7.933333   7.800000  0.973073   
23   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  23   2.966667   3.266667  0.902176   
24   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   0   1.733333   1.900000  0.905701   
25   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   1   0.900000   0.766667  0.998321   
26   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   2   0.433333   0.500000  0.864531   
27   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   3   0.633333   0.400000  0.999999   
28   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   4   0.766667   0.200000  1.000000   
29   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   5   1.266667   0.800000  0.999999   
30   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   6   2.533333   1.566667  0.999999   
31   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   7  10.033333   4.233333  1.000000   
32   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   8  15.533333  13.733333  0.996209   
33   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   9  13.333333   6.400000  1.000000   
34   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  10   7.800000   5.266667  0.999995   
36   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  12   9.700000   7.600000  0.999766   
42   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  18  21.366667  22.800000  0.925854   
43   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  19  15.400000  15.066667  0.975037   
44   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  20  11.600000  11.700000  0.961214   
45   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  21   8.366667   8.666667  0.945913   
46   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  22   5.700000   6.033333  0.931589   
47   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  23   2.800000   2.900000  0.945999   
48   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   0   1.333333   1.233333  0.990563   
49   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   1   0.900000   0.866667  0.980649   
52   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   4   0.166667   0.066667  1.000000   
53   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   5   0.733333   0.300000  1.000000   
54   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   6   1.400000   0.566667  1.000000   
55   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   7   5.866667   3.433333  1.000000   
57   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   9   7.766667   3.900000  1.000000   
58   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  10   5.433333   4.166667  0.999862   
59   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  11   5.566667   5.100000  0.992158   
60   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  12   5.266667   4.366667  0.999097   
61   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  13   5.933333   5.133333  0.997628   
66   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  18  10.333333  10.700000  0.946138   
67   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  19   8.800000   8.466667  0.980928   
68   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  20   6.733333   7.233333  0.920976   
69   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  21   5.100000   5.633333  0.899721   
70   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  22   3.733333   3.700000  0.969672   
71   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  23   1.833333   2.066667  0.883487   
72   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   0   2.600000   2.300000  0.996164   
7

In [29]:
len(w)

199

In [30]:
# FP

len(w[w['균형'] == '균형'])

111

In [31]:
# TN

len(w[w['균형'] == '불균형'])

88

- 불균형인데 균형이라고 판단한 수는 111
- 불균형인데 불균형이라고 판단한 수는 88

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [32]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.4236111111111111

In [33]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.23448275862068965

In [34]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.38202247191011235